### ISSUE TITLE
Trace: Error: Unable to acquire a connection

### ISSUE THREAD

**Issue by Cartoon-Dude** [louislam/uptime-kuma#3478]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### Description

Kuma crashed and the log suggested to report here:
 Trace: Error: Unable to acquire a connection 
2023-07-25T05:00:03+02:00 [MONITOR] ERROR: Please report to https://github.com/louislam/uptime-kuma/issues

### 👟 Reproduction steps

I don't know how it happened, just stopped during the night

### 👀 Expected behavior

Not crashing during the night

### 😓 Actual Behavior

It crashed for no apparent reason 

### 🐻 Uptime-Kuma Version

1.22.1

### 💻 Operating System and Arch

Flatcar Container Linux stable 3510.2.5

### 🌐 Browser

Firefox

### 🐋 Docker Version

20.10.23

### 📝 Relevant log output

```shell
Trace: Error: Unable to acquire a connection
    at Client_SQLite3.acquireConnection (/app/node_modules/knex/lib/client.js:303:13)
    at Runner.ensureConnection (/app/node_modules/knex/lib/execution/runner.js:287:46)
    at Runner.run (/app/node_modules/knex/lib/execution/runner.js:30:30)
    at QueryBuilder_SQLite3.Target.then (/app/node_modules/knex/lib/builder-interface-augmenter.js:24:43)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at Timeout.safeBeat (/app/server/model/monitor.js:857:25)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
2023-07-25T05:00:03+02:00 [MONITOR] ERROR: Please report to https://github.com/louislam/uptime-kuma/issues
```

**Comment by CommanderStorm** [comment]
Just checking the obvious:
The disk you are running on is still alive, correct?

Given that it pretty much died at 05:00 your time, are there any batch jobs running at that time?

**Comment by Cartoon-Dude** [comment]
This machine has just docker for Kuma and Caddy.
I noticed the hosting restarted the machine at 5:00, could it cause that problem? It has --restart=always as argument.

**Comment by CommanderStorm** [comment]
> noticed the hosting restarted the machine at 5:00, could it cause that problem?

That would be my guess, in combination with some sort of IO problem afterwards.
Did restarting the pod fix it?

About the IO:
How many monitors have you set up?
What is your retention period?
(Just a hunch: Are you out of storage?)

**Comment by Cartoon-Dude** [comment]
Yes, everything is fine after restarting the pod.

There is 78 monitors now. The disk space used is 1.5 GB/30
Retention is 180 days

**Comment by CommanderStorm** [comment]
Nothing out of the ordinary, then.
What is your IO type? If you are running on a slow disk this might be the cause of this timeout.

This ***might*** be fixed by a bunch of performance-fixes going into the [`1.23`-release](https://github.com/louislam/uptime-kuma/milestone/34) made by @chakflying 

I have no way of reproducing this issue at the moment.
Given that this is the only report of this issue to date and that we have not made any specific changes to the underlying storage system in the last release
=> I would propose we close this issue and only investigate this further if this happens again.

**Comment by Cartoon-Dude** [comment]
Hello
I asked the hosting company and indeed it is confirmed there was some problems with the storage I/O, it is now resolved and not caused any further problem. This ticket can be now closed.
Thanks for the help.
